x-pack/solutions/security/plugins/security_solution/public/common/hooks/use_ai_connectors.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo, useCallback } from 'react';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { useAppToasts } from './use_app_toasts';
+import { loadAiConnectors } from '../utils/connectors/ai_connectors';
+import * as i18n from './translations';
+import { useKibana } from '../lib/kibana';
+
+const QUERY_KEY = ['ai_connectors'];
+
+export const useAIConnectors = () => {
+  const {
+    services: { http },
+  } = useKibana();
+  const { addError } = useAppToasts();
+  const { data, ...rest } = useQuery({
+    queryKey: QUERY_KEY,
+    queryFn: () => loadAiConnectors(http),
+    onError: (error) => {
+      addError(error, {
+        title: i18n.ERROR_FETCH_AI_CONNECTORS,
+      });
+    },
+    refetchOnWindowFocus: false,
+  });
+
+  return useMemo(
+    () => ({
+      aiConnectors: data ?? [],
+      ...rest,
+    }),
+    [data, rest]
+  );
```

Comment:
The destructed `rest` variable will be a new object on every run, even when `useQuery` returns the same object reference. So, it looks like this hook is returning a new object every time, even when nothing has changed, and useQuery returned a cached result.

Could we change that and put the entire useQuery `result` as the `useMemo` dependency instead, and destruct inside? Or otherwise, destruct to all the single properties we need (`isLoading`, `error`... ), instead of using the spread (`...rest`) that creates a new object?